npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -354,8 +354,8 @@ def self.run_pnpm_command(command, fingerprint: nil)
           package_manager_run_command(PNPMPackageManager::NAME, command, fingerprint: fingerprint)
         else
           Dependabot::SharedHelpers.run_shell_command(
-            "pnpm #{command}",
-            fingerprint: "pnpm #{fingerprint || command}"
+            "pnpm --config.manage-package-manager-versions=0 #{command}",
```

Comment:
i assume this is what disables `packageManager` ? 

i'm not sure that's a good idea.
it has the potential to break all projects relying on an older version of pnpm.
if my `package.json` contains a `packageManager` field - i'd expect dependabot to respect this.
Everything else could lead to breaking updates (it'll potentially update `pnpm-lock.json` to a non-supported format by the `packageManager` selected) - and would make the current issue worse - as it'll take away the possibility to work around it by pinning a lower pnpm version as part of `package.json`.

Reply:
Prior to dependabot updating to pnpm v10 the only time pnpm's builtin `packageManager` support would have kicked in is when project specific configuration (e.g. in `.npmrc`) enabled the feature. If `packageManager` worked previously it would be due to dependabot's builtin `corepack` support (handled in the `if` branch) or via explicit opt-in.

My implementation here does break one scenario. Because it is set via CLI it will override explicit opt-in in project configuration. A better place to have this set would be in `~/.config/pnpm/rc`, so it can still be overridden. Failing that it could be conditionally added by checking with `pnpm config get manage-package-manager-versions`, where an `"undefined"` result would indicate the user has not opted-in.